src/fides/api/service/messaging/messaging_crud_service.py

```diff
@@ -80,32 +88,321 @@ def get_messaging_config_by_key(db: Session, key: FidesKey) -> MessagingConfigRe
     )
 
 
-def get_all_messaging_templates(db: Session) -> List[MessagingTemplate]:
-    # Retrieve all templates from the database
-    templates_from_db = {
-        template.key: template.content for template in MessagingTemplate.all(db)
-    }
-
-    # Create a list of MessagingTemplate models, using defaults if a key is not found in the database
+def get_all_basic_messaging_templates(db: Session) -> List[MessagingTemplate]:
+    """
+    Retrieve all templates from the database, filling in default values if any default template type
+    is not found in the database.
+    """
     templates = []
-    for key, template in DEFAULT_MESSAGING_TEMPLATES.items():
-        content = templates_from_db.get(key, template["content"])
-        templates.append(
-            MessagingTemplate(
-                key=key,
-                content=content,
+    for template_type, template in DEFAULT_MESSAGING_TEMPLATES.items():
+        template_from_db = _basic_messaging_template_by_type(db, template_type)
+        if template_from_db:
+            templates.append(
+                MessagingTemplate(
+                    type=template_type,
+                    content=template_from_db.content,
+                )
+            )
+        else:
+            templates.append(
+                MessagingTemplate(
+                    type=template_type,
+                    content=template["content"],
+                )
             )
-        )
 
     return templates
 
 
-def get_messaging_template_by_key(db: Session, key: str) -> Optional[MessagingTemplate]:
-    template = MessagingTemplate.get_by(db, field="key", value=key)
+def _basic_messaging_template_by_type(
+    db: Session, template_type: str
+) -> Optional[MessagingTemplate]:
+    """
+    Provide a consistent way to retrieve basic messaging template given a type.
+
+    Scenario A: One template configured with type
+    Result: Return that template
+
+    Scenario B: Multiple templates configured with type, none with default property
+    Result: Return the first template
+
+    Scenario C: Multiple templates configured with type, one with default property
+    Result: Return the template with default property
+    """
+    template = None
+    templates = (
+        MessagingTemplate.query(db=db)
+        .filter(MessagingTemplate.type == template_type)
+        .all()
+    )
+
+    if len(templates) == 1:
+        template = templates[0]
+    elif len(templates) > 1:
+        default_property = Property.get_by(db=db, field="is_default", value=True)
+        if default_property:
+            template = MessagingTemplate.filter(
+                db=db,
+                conditions=(
+                    (MessagingTemplate.type == template_type)
+                    & (
+                        MessagingTemplate.properties.any(
+                            Property.id == default_property.id
+                        )
+                    )
+                ),
+            ).first()
```

Comment:
A couple things, this should also be ordered - what is "first" can change.  

Second, good job figuring out this query.  This is a little less efficient than it could be, SQLAlchemy is joining in your Manytomany table behind the scenes  "MessagingTemplateToProperty" and then doing a "Exists Select 1)...

I think it's a little clearer to join in the many to many table directly, something like:

```python
db.query(MessagingTemplate).join(MessagingTemplateToProperty).filter(MessagingTemplate.type==template_type, MessagingTemplateToProperty.property_id==default_property_id).order_by(MessagingTemplate.created_at.first()),
```

or however you choose to order.

In both of these queries, there's some interesting SQLAlchemy magic happening around where you defined the relationships to join the right columns together!

Reply:
oh interesting, I didn't know that SQLalchemy would use the `MessagingTemplateToProperty` table here. Thanks for pointing this out! Do you have any additional resources for learning how SQLAlchemy emits queries behind the scenes? I have a lot to learn about SQLAlchemy overall.

Reply:
you can wrap a query object that's ready to be executed but not actually executing in a print statement! for example(db.query().filter()...) drop the .first() or .all() -

Reply:
ohhh thank you!